src/api-service/__app__/timer_workers/__init__.py

```diff
@@ -6,43 +6,44 @@
 import logging
 
 import azure.functions as func
-from onefuzztypes.enums import ScalesetState
+from onefuzztypes.enums import NodeState, PoolState
 
+from ..onefuzzlib.autoscale import autoscale_pool
 from ..onefuzzlib.dashboard import get_event
-from ..onefuzzlib.pools import Scaleset
+from ..onefuzzlib.orm import process_update
+from ..onefuzzlib.pools import Node, Pool, Scaleset
 
 
 def process_scaleset(scaleset: Scaleset) -> None:
     logging.debug("checking scaleset for updates: %s", scaleset.scaleset_id)
 
-    if scaleset.state == ScalesetState.resize:
-        scaleset.resize()
-
     # if the scaleset is touched during cleanup, don't continue to process it
     if scaleset.cleanup_nodes():
         logging.debug("scaleset needed cleanup: %s", scaleset.scaleset_id)
         return
 
-    if (
-        scaleset.state in ScalesetState.needs_work()
-        and scaleset.state != ScalesetState.resize
-    ):
-        logging.info(
-            "exec scaleset state: %s - %s",
-            scaleset.scaleset_id,
-            scaleset.state,
-        )
-
-        if hasattr(scaleset, scaleset.state.name):
-            getattr(scaleset, scaleset.state.name)()
-        return
+    process_update(scaleset)
 
 
 def main(mytimer: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
+    Node.mark_oudated_nodes()
+    nodes = Node.search_states(states=NodeState.needs_work())
```

Comment:
These two lines is result is two queries to the node table. There will be another query to get the dead nodes too. 
it think it is worth getting all the nodes in one query and process them here in the service once.

Reply:
I will leave this alone for now, and will address this when adding the heartbeat checking.  (See discussion in Teams regarding this)